src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/DiagnosticTests.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Razor.IntegrationTests;
+
+public class DiagnosticTests : AbstractRazorEditorTest
+{
+    [IdeFact]
+    public async Task Diagnostics_ShowErrors()
+    {
+        // Arrange
+        await TestServices.SolutionExplorer.OpenFileAsync(RazorProjectConstants.BlazorProjectName, RazorProjectConstants.CounterRazorFile, ControlledHangMitigatingCancellationToken);
+
+        await TestServices.Editor.SetTextAsync(@"
+<PageTitle>
+
+@code{
+    public void Function(){
+        return """"
+    }
+}
+", ControlledHangMitigatingCancellationToken);
+
+        // Act
+        var errors = await TestServices.ErrorList.WaitForErrorsAsync("Counter.razor", ControlledHangMitigatingCancellationToken);
+
+        // Assert
+        Assert.Collection(errors,
```

Comment:
Can we put a HTML diagnostic in here and well?

Reply:
I can include an HTML diagnostic, but it will be coming from the Razor-side. As written this won't go out to the HTML LangaugeServer. This is in keeping with the way the [HTMLCSharp LS handles things today](https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DocumentPullDiagnosticsHandler.cs#L119-L135). Note that while that call would try to got to the HTML LS it won't ever find anything because it's been given the C# URI.

If we want to start including HTML diagnostics from a LS I propose we do that as a separate issue.

Reply:
Since this is an integration test, I don't think that distinction matters, but I'm fine with a follow up... BUT it would be confidence building to be sure this PR doesn't break Html diagnostics somehow. Up to you though, if you'd rather do it in a follow up.